### PR TITLE
MULE-14764: HttpRequester lifecycle is not managed properly in OAuth scenarios.

### PR DESCRIPTION
**Pull request by facundovs**

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpConnectorMessageProcessorProvider.java [hunk 1]
```diff
@@ -113,4 +114,11 @@ public void dispose()
     {
         cachedMessageProcessors.invalidateAll();
     }
+
+    @Override
+    public void stop() throws MuleException
+    {
+        cachedMessageProcessors.invalidateAll();
```
Comment (afelisatti) [inline modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpConnectorMessageProcessorProvider.java]: If this is the same as `dispose` why have both?

File: modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpRequesterBuilder.java [hunk 2]
```diff
@@ -162,6 +162,10 @@ else if (httpRequester.getConfig() == null)
                     }
                 }
             }
+            else
+            {
+                requestConfig.start();
```
Comment (afelisatti) [inline modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpRequesterBuilder.java]: Is there a chance you'll start it several times?

File: modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpConnectorMessageProcessorProvider.java (hunk 1 above)
Comment (facundovs) [reply modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpConnectorMessageProcessorProvider.java]: The issue ocurrs when the application is stopped from ARM.
In this case the dispose phase is not executed.

File: modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpRequesterBuilder.java (hunk 2 above)
Comment (facundovs) [reply modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpRequesterBuilder.java]: This scenario is covered by the start method code:

https://github.com/mulesoft/mule/blob/mule-3.x/modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequesterConfig.java#L340

File: modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpConnectorMessageProcessorProvider.java (hunk 1 above)
Comment (afelisatti) [reply modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpConnectorMessageProcessorProvider.java]: Yes, but you have some other scenarios to consider:
1) The component is stopped and an undeploy occurs, you'd invalidate the cache again.
2) The component is up and running and an undeploy occurs, you'd invalidate the cache twice.

**Pull request merged**